api/src/main/java/io/strimzi/api/kafka/model/template/StatefulSetTemplate.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.template;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.Serializable;
+
+/**
+ * Representation of a Pod template for Strimzi stateful sets.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "metadata"})
+public class StatefulSetTemplate implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private MetadataTemplate metadata;
+
+    @Description("Metadata which should be applied to the stateful set.")
```

Comment:
I think we should use `StatefulSet` and likewise for other k8s kinds. We can use asciidoc format here, remember.